api/python/cellxgene_census/src/cellxgene_census/_open.py

```diff
@@ -68,38 +71,52 @@ def _open_soma(
     Private. Merge config defaults and return open census as a soma Collection/context.
     """
 
+    # if no user-defined context, cellxgene_census defaults take precedence over SOMA defaults
+    context = context or get_default_soma_context()
+
+    # If S3 provider is specified, build a default context with a region
     if locator["provider"] == "S3":
-        context = _build_soma_tiledb_context(locator.get("region"), context)
-    else:  # If no provider is specified, build a default context (don't pass region)
-        context = _build_soma_tiledb_context(None, context)
+        context = context.replace(tiledb_config={"vfs.s3.region": locator.get("region")})
 
     return soma.open(locator["uri"], mode="r", soma_type=soma.Collection, context=context)
 
 
-def _build_soma_tiledb_context(
-    s3_region: Optional[str] = None, context: Optional[soma.options.SOMATileDBContext] = None
-) -> soma.options.SOMATileDBContext:
-    """
-    Private. Build a SOMATileDBContext with sensible defaults. If user-defined context is provided, only update the
-    `vfs.s3.region` only.
+def get_default_soma_context() -> soma.options.SOMATileDBContext:
+    """Return a ``SOMATileDBContext`` with sensible defaults that can be further customized by the user.  The customized
+    context can then be passed to ``open_soma()`` or a ``SOMAObject.open()`` method,
+    such as ``tiledbsoma.Experiment.open()``.  Use the ``replace()`` method on the returned object to customize its
+    settings.
+
+    Returns:
+        A ``SOMATileDBContext`` object with sensible defaults.
+
+    Examples:
+
+        To reduce the amount of memory used by TileDB-SOMA I/O operations:
+
+        .. highlight:: python
+        .. code-block:: python
+
+            ctx = cellxgene_census.get_default_soma_context().replace(
+                tiledb_config={"py.init_buffer_bytes": 128 * 1024**2,
```

Comment:
why not just allow the user to pass a `config` to the `open_soma()` call? That makes it closer to the tiledbsoma syntax, and I think easier to use.

Example:
```
census = cellxgene_census.open_soma(
    uri="s3://my-private-bucket/census/soma",
    tiledb_config={"vfs.s3.region": "us-east-1"}
)
```

Reply:
`TileDBObject.open()` calls still just take a `context`, afaict (https://github.com/single-cell-data/TileDB-SOMA/blob/24b57de4fad1828d4fdcda99ca630bf048b22cd6/apis/python/src/tiledbsoma/_tiledb_object.py#L44).  But the `SOMATileDBContext` object does now accept a config dict.  So for your suggestion, would it make sense to accept a `tiledb_config` in the `get_default_soma_context()` call instead? My thinking is that if the design of `SOMATileDBContext` changes, the `open_soma()` method doesn't have to change, though the get_default_soma_context() would.  But since it's more "context savvy", maybe that's a little better?

Reply:
The other consideration is that `get_default_soma_context()` is also useful when opening `TileDBObject`s directly, and wanting to use the Census's context defaults. So if we eliminated the method in favor of _only_ supporting context overrides via `open_soma()`, this use case has to resort back to specifying _all_ the context settings correctly.

Reply:
I'm not suggesting one over the other, but doing both.  Common case is `open_soma` while overriding one param such as memory.  But you are right that the `get context` is also useful.

So, I'd do three things:
1. add a `tiledb_config={}` argument to the `open_soma` call.
2. retain the newly added get_default_soma_context
3. add the same `tiledb_confg` argument to get_default_soma_context

I'm primarily suggesting that the dict-override is superior, and should be pulled through our API

Reply:
As discussed separately, will merge this and implement this suggestion on a new PR.